modules/common/util/template_util.go

```diff
@@ -157,6 +157,8 @@ func ExecuteTemplateData(templateData string, data interface{}) (string, error)
 		"add":   add,
 		"lower": lower,
 	}
+	// remove extra newlines
+	templateData = strings.ReplaceAll(templateData, "\n\n\n", "")
```

Comment:
yeah this potentially remove all the new lines between two lines of config making the config invalid. 

I think what we need is replacing every two or more consecutive new lines with a single one until we have no more two consecutive new lines.

Reply:
Also this only applies the transformation on the template not on the rendered text. The rendered text ends up in the buff variable after tmpl.Execute()

Reply:
Updated outside execute.